internal/controllers/machinedeployment/machinedeployment_controller_test.go

```diff
@@ -264,10 +265,40 @@ func TestMachineDeploymentReconciler(t *testing.T) {
 		}, timeout).Should(BeEquivalentTo(desiredMachineDeploymentReplicas))
 
 		//
-		// Update a MachineDeployment, expect Reconcile to be called and a new MachineSet to appear.
+		// Update the InfraStructureRef of the MachineDeployment, expect Reconcile to be called and a new MachineSet to appear.
 		//
-		t.Log("Setting a label on the MachineDeployment")
-		modifyFunc = func(d *clusterv1.MachineDeployment) { d.Spec.Template.Labels["updated"] = "true" }
+
+		t.Log("Updating the InfrastructureRef on the MachineDeployment")
+		// Create the InfrastructureTemplate
+		// Create infrastructure template resource.
+		infraResource2 := map[string]interface{}{
+			"kind":       "GenericInfrastructureMachine",
+			"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
+			"metadata":   map[string]interface{}{},
+			"spec": map[string]interface{}{
+				"size": "5xlarge",
+			},
+		}
+		infraTmpl2 := &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"spec": map[string]interface{}{
+					"template": infraResource2,
+				},
+			},
+		}
+		infraTmpl2.SetKind("GenericInfrastructureMachineTemplate")
+		infraTmpl2.SetAPIVersion("infrastructure.cluster.x-k8s.io/v1beta1")
+		infraTmpl2.SetName("md-template-2")
+		infraTmpl2.SetNamespace(namespace.Name)
```

Comment:
Can we inline those into the unstructured?

I would also recommend to define the whole unstructured where we define infraTmpl just to get it out of the logic of the test code